### PR TITLE
use bash from env instead of fixed path, check version

### LINKED ISSUE #30: macOS: ocrd-olena-binarize fails (old bash version)
macOS provides an old version of bash which obviously does not support `declare -A`:

    $ ocrd-olena-binarize --help
    /Users/venv-20200202/bin/ocrd-olena-binarize: line 34: declare: -A: invalid option
    declare: usage: declare [-afFirtx] [-p] [name[=value] ...]

    $ /bin/bash --version
    GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin19)
    Copyright (C) 2007 Free Software Foundation, Inc.

### PR DESCRIPTION
**Pull request by bertsky**
Fixes #30

### CODE REVIEW

**Review by kba — APPROVED** [review]

File: ocrd-olena-binarize [hunk 1]
```diff
@@ -24,7 +24,8 @@ set -o pipefail
 # by inserting an AlternativeImage to the PAGE-XML.
 # Regardless, adds new image and new PAGE-XML to METS.
 
-which ocrd >/dev/null 2>/dev/null || { echo "ocrd not in \$PATH. Panicking"; exit 1; }
+which ocrd >/dev/null 2>/dev/null || { echo >&2 "ocrd not in \$PATH. Panicking"; exit 1; }
+((BASH_VERSINFO[0]<4 || BASH_VERSINFO[1]<4)) && echo >&2 "bash $BASH_VERSION is too old. Panicking" && exit 1
```
Comment (stweil) [inline ocrd-olena-binarize]: Travis fails here:

    bash 4.3.48(1)-release is too old. Panicking

What do you want to check? I'm afraid that even a recent bash fails now:

    $ ((BASH_VERSINFO[0]<4 || BASH_VERSINFO[1]<4)) && echo >&2 "bash $BASH_VERSION is too old. Panicking"
    bash 5.0.11(1)-release is too old. Panicking

Comment (stweil) [reply ocrd-olena-binarize]:
```suggestion
((BASH_VERSINFO<4 || BASH_VERSINFO==4 && BASH_VERSINFO[1]<4)) && echo >&2 "bash $BASH_VERSION is too old. Please install bash 4.4 or newer." && exit 1
```

**Review by stweil — CHANGES_REQUESTED** [review]

File: ocrd-olena-binarize (hunk 1 above)
Comment (bertsky) [reply ocrd-olena-binarize]: > Travis fails here:
> 
> ```
> bash 4.3.48(1)-release is too old. Panicking
> ```

I'm afraid we'd have to abandon this Travis configuration. 4.4 is a sane minimum (see disussion in the issue)

**Review by stweil — APPROVED** [review]

**Comment by stweil** [comment]
We already suggest Ubuntu 18.04 which provides bash 4.4, so we should use that for CI, too. https://docs.travis-ci.com/user/reference/bionic/ explains how that works. See PR #32.

**Comment by bertsky** [comment]
> We already suggest Ubuntu 18.04 which provides bash 4.4, so we should use that for CI, too. https://docs.travis-ci.com/user/reference/bionic/ explains how that works. See PR #32.

Same thought, only I was slightly faster :-)

**Comment by stweil** [comment]
@bertsky, it would be nice to have the problem with case insensitive filesystems (macOS!) fixed in the next release, too. That requires an update of OCR-D/assets with https://github.com/OCR-D/assets/pull/70.

**Comment by bertsky** [comment]
> it would be nice to have the problem with case insensitive filesystems (macOS!) fixed in the next release, too. That requires an update of OCR-D/assets with [OCR-D/assets#70](https://github.com/OCR-D/assets/pull/70).

@stweil sure! But are the assets ready yet? When OCR-D/assets#70 is merged, we can start a PR with the respective update of `repo/assets` here. (Or even against the PR branch, if you want to verify this works here before finalizing in assets.)

**Comment by stweil** [comment]
> But are the assets ready yet?

Yes, I think so.

**Comment by stweil** [comment]
The master branch of OCR-D/assets includes two commits from @kba which would then also be part of the update for ocrd_olena, so we need his judgement, too.

**Comment by bertsky** [comment]
> The master branch of assets includes two commits from @kba which would then also be part of the update for ocrd_olena, so we need his judgement, too.

I cannot see any of that. Your [stweil:duplicates](https://github.com/stweil/assets/tree/duplicates) currently says:
> This branch is 2 commits ahead of OCR-D:master. 

(i.e. nothing behind)

Anyway, this should be discussed in a new PR. Since Travis succeeded, I'll merge this now.

**Comment by kba** [comment]
> Anyway, this should be discussed in a new PR

I'll merge the deduplication PR in assets today, just need to make sure tests run through for the projects that depend on it.

**Pull request merged**